cmd/launcher/updater/updater_test.go

```diff
@@ -0,0 +1,226 @@
+package updater
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/cmd/launcher/updater/mocks"
+	"github.com/kolide/updater/tuf"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func Test_updaterCmd_execute(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		updater                 *mocks.Updater
+		stopChan                chan bool
+		config                  *UpdaterConfig
+		runUpdaterRetryInterval time.Duration
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		// in this test, the calls to run are the only thing we can really assert against
+		// leave this field empty and the test will fail if there is a call to run function made
+		// add 3 funcs here and the test will expect updater.Run() to be called 3 times
+		updaterRunReturns []func(opts ...tuf.Option) (stop func(), err error)
+		callStopChanAfter time.Duration
+		assertion         assert.ErrorAssertionFunc
+	}{
+		{
+			name: "success",
+			fields: fields{
+				updater: &mocks.Updater{},
+				config: &UpdaterConfig{
+					Logger: log.NewNopLogger(),
```

Comment:
Since there's now a `Create` function, I'd probably take these out of the tests, and do something like:
```
if config.logger == nil {
  config.logger = log.NewNopLogger()
}
```

Probably pairs well with the `log.With` that's there

Reply:
If I'm understanding correctly, the suggestion is to instantiate the updater via a call to an create/new function ...
I think to do that would require a bit more of a refactor. Currently, the create/new function for `cmd/launcher/updater` is creating a new instance of `pkg/autoupdate`. I think it might be better to pass an instance of `pkg/autoupdate` to the `cmd/launcher/updater` create/new function. This way we can mock that dependency for the `cmd/launcher/update` tests. I suppose somewhere around [here](https://github.com/kolide/launcher/blob/master/cmd/launcher/launcher.go#L188) we could create the new instance of `pkg/autoupdate` and pass it to `cmd/launcher/updater`

Reply:
updated the create function, but not updating the tests per our conversation as it would require further refactor